app/src/main/java/com/example/myapplication/FirstFragment.java

```diff
@@ -80,8 +80,29 @@ public void afterTextChanged(Editable s) {
                 if (binding.input.getText() == null){
                     return;
                 }
-                MainActivity.teamNumber = String.valueOf(binding.input.getText());
-                UserModel.getMatchData().setTeamNumber(Integer.parseInt(MainActivity.teamNumber));
+                String teamNumber = String.valueOf(binding.input.getText());
+                UserModel.getMatchData().setTeamNumber(Integer.parseInt(teamNumber));
+            }
+        });
+
+        binding.matchInput.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                if (binding.matchInput.getText() == null){
+                    return;
+                }
+                String matchNumber = String.valueOf(binding.matchInput.getText());
+                UserModel.getMatchData().setMatchNumber(Integer.parseInt(matchNumber));
```

Comment:
If the text is empty (i.e., someone types a number, then hits backspace), the call to parseInt will fail and the app will crash. Consider checking that the text is not empty (in addition to the null check done above)